src/entities/activity/ui/items/ActionPlan/ResponseSegment.tsx

```diff
@@ -118,32 +126,52 @@ export const ResponseSegment = ({ phrasalData, field, isAtStart }: ResponseSegme
 
   return (
     <Text component="span" fontWeight="700" fontSize="inherit" lineHeight="inherit">
-      {fieldDisplayMode === 'bullet_list' ||
-      fieldDisplayMode === 'bullet_list_option_row' ||
-      fieldDisplayMode === 'bullet_list_text_row' ? (
-        <>
-          {isAtStart ? null : (
-            <span>
-              &nbsp;
-              <br />
-            </span>
-          )}
-          <Box
-            component="ul"
-            marginTop={isAtStart ? `0px` : undefined}
-            paddingLeft={`${listPadding}px`}
-          >
-            {words.map((item, index) => (
-              <li key={index}>{item}</li>
-            ))}
-          </Box>
-        </>
-      ) : (
-        <>
-          {isAtStart ? '' : ' '}
-          {joinSentenceWords(words)}
-        </>
-      )}
+      {(() => {
+        if (
+          fieldDisplayMode === 'bullet_list' ||
+          fieldDisplayMode === 'bullet_list_option_row' ||
+          fieldDisplayMode === 'bullet_list_text_row'
+        ) {
+          return (
+            <>
+              {isAtStart ? null : (
+                <span>
+                  &nbsp;
+                  <br />
+                </span>
+              )}
+              <Box
+                component="ul"
+                marginTop={isAtStart ? `0px` : undefined}
+                paddingLeft={`${listPadding}px`}
+              >
+                {words.map((item, index) => (
+                  <li key={index}>{item}</li>
+                ))}
+              </Box>
+            </>
+          );
+        } else if (fieldPhrasalDataType === 'paragraph') {
```

Comment:
I see why you created a closure here, because of the extra condition and wanting to avoid nested ternaries. However, you don't need to go to this trouble. It looks like me like Billie created the `joinSentenceWords` function as a variable for the very purpose of customizing it to do different things based on different item response types. So it would be easier to read, and a smaller change, if you just did this:

[Line 61](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/542/files#diff-ab9d6cf61610fc5f6da132ec966f87b80d8bf58ed8bdcbd1c60c17af110eec3dL61):
```tsx
  } else if (fieldPhrasalData.context.itemResponseType === 'paragraphText') {
    joinSentenceWords = (values) =>
      values.map((item, index) => <div key={index}>{item || ' '}</div>);
  }
```
(And to repeat from my last comment, the `' '` above is a non-breaking space.)

Then you can revert [this entire change (lines 121-146)](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/542/files#diff-ab9d6cf61610fc5f6da132ec966f87b80d8bf58ed8bdcbd1c60c17af110eec3dR129-R174).